src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -266,7 +298,8 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                     }
 
                                     // ensure there are no duplicate event ids.
-                                    if (!eventIds.Add(lm.EventId))
+                                    // LoggerMessageAttribute has constructors that don't take an EventId, we need to exclude the default Id -1 from duplication checks.
```

Comment:
We also talked about generating an id based on the EventName if no id was provided.  Are we going to do that?

Reply:
I'm going to do that in the generator in dotnet/extensions. Not sure if its worth doing in the runtime generator.

Reply:
If we're still having to duplicate the generators, I'd like to minimize the differences between them to only what's truly necessary.

Reply:
I can address this. I am not seeing many users providing Event name with logging though.

Reply:
> So @tarekgh, we should make sure both generators do this the same way. So please let me know what hash function you end up using on the event name in order to produce the event id, and I'll use the same thing on my end.

I am thinking to copy the non-randomized hashing from https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs,807

Reply:
for the record, I opened the PR https://github.com/dotnet/runtime/pull/87892